gulpfile.js

```diff
@@ -40,16 +51,127 @@
     );
 
     gulp.task(
-        'less',
+        'build:clean',
+        function (callback) {
+            rimraf('public/css', callback);
+        }
+    );
+
+    gulp.task(
+        'build:less',
         function () {
             return gulp
                 .src('public/less/main.less')
-                .pipe(less({ paths: [ __dirname + '/public/less' ] }))
+                .pipe(less({
+                    paths: [ 'public/less' ]
+                }))
                 .pipe(gulp.dest('public/css'));
         }
     );
 
     gulp.task(
+        'build',
+        [
+            'build:clean',
+            'build:less'
+        ]
+    );
+
+    gulp.task(
+        'package:clean',
+        function (callback) {
+            if (environment === 'development') {
+                throw new Error('Cannot use "package" tasks in development environment');
+            }
+            rimraf('dist/' + environment, function () {
+                mkdirp('dist/' + environment, callback);
+            });
+        }
+    );
+
+    gulp.task(
+        'package:less',
+        function () {
+            if (environment === 'development') {
+                throw new Error('Cannot use "package" tasks in development environment');
+            }
+            return gulp
+                .src('public/less/main.less')
+                .pipe(less({
+                    paths: [ 'public/less' ],
+                    optimize: true
+                }))
+                .pipe(gulp.dest('dist/' + environment));
+        }
+    );
+
+    gulp.task(
```

Comment:
This task is the core of the pull request.  It uses `r.js` the RequireJS optimiser, to generate `application.js`.